### PR TITLE
Added schema validations

### PR DESCRIPTION
**Pull request by bzsurbhi**
#### Description of changes

This PR adds below mentioned schema validations:

1) Auth must be annotated with model
2) @belongsTo fields type must match the type of primary key in related model
3) Every @manyToMany relation has correct types
4) Relation name does not conflict with an existing type name
5) '_version', '_changedAt', '_deleted' are reserved names and cannot be used as field names
6) 'Query', 'Mutation', 'Subscription' a reserved names and cannot be used as type names
7) Sort key fields value exists in the model

It also updates the graphql from 16.6 to 15.5 as recommended in this PR https://github.com/aws-amplify/amplify-category-api/pull/1300 

##### CDK / CloudFormation Parameters Changed

None

#### Issue #, if available

N/A

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-schema-validator/src/helpers/util.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const getGraphqlName = (val: string): string => {
```
[the remark this answers is not in the document]
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/helpers/util.ts:7]: Added

File: packages/amplify-schema-validator/src/validators/sort-key-field-exists.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+} from 'graphql';
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+
+/**
+   * Validates that sort key fields exists in model
+   *
+   * @param schema graphql schema
+   * @returns true if sort key fields exists in model
+   */
+export const validateSortKeyFieldsExistsInModel = (schema: DocumentNode): Error[] => {
```
[the remark this answers is not in the document]
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/sort-key-field-exists.ts]: Done

File: packages/amplify-schema-validator/package.json [hunk 3]
```diff
@@ -27,7 +27,9 @@
     "ts-node": "^8.5.0"
   },
   "dependencies": {
-    "graphql": "^16.6.0"
+    "aws-sdk": "^2.1141.0",
+    "graphql": "^15.5.0",
+    "npm-check-updates": "^16.1.0"
```
Comment (alharris-at) [inline packages/amplify-schema-validator/package.json:32]: Why the change to `aws-sdk` and `npm-check-updates`?

Comment (bzsurbhi) [reply packages/amplify-schema-validator/package.json]: reverted the changes

**Review by alharris-at — APPROVED** [review]

File: packages/amplify-schema-validator/package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "main": "./dist/index.js",
   "private": true,
   "types": "./dist/index.d.ts",
-  "version": "0.2.1",
+  "version": "1.22.0",
```
Comment (sundersc) [inline packages/amplify-schema-validator/package.json:6]: Is this intentional?

Comment (bzsurbhi) [reply packages/amplify-schema-validator/package.json]: Reverted

**Review by sundersc — APPROVED** [review]

**Review by alharris-at — APPROVED** [review]
